media-api/app/lib/ImageResponse.scala

```diff
@@ -320,6 +324,28 @@ object ImageResponse {
   private def hasExports(image: Image) = image.exports.nonEmpty
 
   private def hasUsages(image: Image) = image.usages.nonEmpty
+
+  def addAliases(config: MediaApiConfig, source: SourceWrapper[Image]): Reads[JsObject] = {
+    @tailrec
+    def nestedLookup(jsLookup: JsLookupResult, pathComponents: List[String]): JsLookupResult = {
+      pathComponents match {
+        case Nil => jsLookup
+        case head :: tail => nestedLookup(jsLookup \ head, tail)
+      }
+    }
+
+    val aliases: Seq[(String, JsValue)] = config.fieldAliasConfigs.flatMap { config =>
+      val parts = config.elasticsearchPath.split('.').toList.filter(_.nonEmpty)
+      val lookupResult = nestedLookup(JsDefined(source.source), parts)
+      lookupResult.toOption.map {
+        config.alias -> _
+      }
+    }
```

Comment:
Fairly minor but it would be pleasant to extract this into another method so that this business logic isn't cluttering this function. Perhaps something like `extractAliasFieldValues(config: MediaApiConfig, source: SourceWrapper[Image]): Seq[(String, JsValue)]` on the `ImageResponse` object.

In fact `addAliases` then becomes `addAliases(aliases: Seq[(String, JsValue)])` and the aliases are actually calculated at the call site around line 89.